app/src/androidTest/java/com/cepheuen/elegantnumberbuttonsample/MainActivityTest.java

```diff
@@ -0,0 +1,50 @@
+package com.cepheuen.elegantnumberbuttonsample;
+
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.matcher.ViewMatchers.assertThat;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.core.AllOf.allOf;
+
+import androidx.test.espresso.Espresso;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class MainActivityTest {
+    @Rule public ActivityTestRule<MainActivity> rule = new ActivityTestRule<>(MainActivity.class,
+            false, true);
+    private MainActivity activity;
+
+    @Before
+    public void get_activity() { activity = rule.getActivity(); }
+
+    @Test
+    public void test_add_button_click() {
+        try {
+            // Change btn1 text to find it with ViewMatchers
+            rule.runOnUiThread(() -> activity.btn1.addBtn.setText("target"));
+        } catch (Throwable throwable) {
+            throwable.printStackTrace();
+        }
+        Espresso.onView(allOf(withId(R.id.add_btn), withText("target"))).perform(click());
+        assertThat(activity.btn1.getNumber(), Matchers.is("1"));
+    }
```

Comment:
We can add tests to cover other functions as well but this should surely be a good place to start.